tests/test_series.py

```diff
@@ -731,6 +731,20 @@ def test_types_group_by_with_dropna_keyword() -> None:
     s.groupby(level=0).sum()
 
 
+def test_types_groupby_iter() -> None:
+    s = pd.Series([1, 1, 2], dtype=int)
+    series_groupby = pd.Series([True, True, False], dtype=bool)
+    first_group = next(iter(s.groupby(series_groupby)))
+    check(
+        assert_type(first_group[0], "bool"),
+        bool,
+    )
+    check(
+        assert_type(first_group[1], "pd.Series[int]"),
+        pd.Series,
```

Comment:
```suggestion
        assert_type(first_group[1], "pd.Series[int]"),
        pd.Series, np.integer
```

You do need quotes here because `pd.Series[int]` is not a type at runtime.